src/core/server/http/router/request.ts

```diff
@@ -186,7 +186,7 @@ export class KibanaRequest<
 
   private getEvents(request: Request): KibanaRequestEvents {
     const finish$ = merge(
-      fromEvent(request.raw.req, 'end'), // all data consumed
```

Comment:
In my testing, this change does appear to be correct. Right now the aborted$ observable will complete prior to the response being sent back, which may cause us to lose aborted events. This appears to only happen on `POST` requests because the `end` event is being fired when the request body is read. On `GET` requests, the aborted$ Observable does not complete until the connection is closed.

I propose we bundle this fix with the addition of a real `completed$` event which will emit whenever the request is either aborted, or a response has been sent back to the client. Here is a PR: https://github.com/elastic/kibana/pull/73898